[tp] improve documentation

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115880

Improve the TP documentation in terms of format and descriptions

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225

wanchaol committed:
[tp] improve documentation

Improve the TP documentation in terms of format and descriptions

[ghstack-poisoned]

XilunWu reviewed (approved):
lgtm.

awgu commented on docs/source/distributed.tensor.parallel.rst:
```diff
@@ -32,7 +32,7 @@ Tensor Parallelism supports the following parallel styles:
 and perform necessary layout redistributions, without distribute the module
 parameters to DTensors, the following classes can be used in
-the ``parallelize_plan``of ``parallelize_module``:
+the `parallelize_plan` of `parallelize_module`:
```
I thought that we need double backtick for code type and that single backtick is italics, but I may be wrong.

wanchaol replied:
It's kind of weird to me too, but the lastest main docs looks wrong format so I switched back to single backtick, see lines before this API https://pytorch.org/docs/main/distributed.tensor.parallel.html#torch.distributed.tensor.parallel.PrepareModuleInput

I'll play around with the code types a bit more and see the refreshed docs before landing

awgu replied:
Maybe it is missing the space between the last backtick of \`\`parallelize_plan\`\` and of, causing it to render improperly?

wanchaol replied:
good point, lm try that instead

wanchaol replied:
adjusted multiple places to double backtick , but the built doc still rendering the old version, gonna land this first and see the doc

wanchaol committed:
Update on "[tp] improve documentation"


Improve the TP documentation in terms of format and descriptions

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

wanchaol committed:
Update on "[tp] improve documentation"


Improve the TP documentation in terms of format and descriptions

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
